environment/environment.go

```diff
@@ -0,0 +1,906 @@
+package environment
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/contracts"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	kubeAppTypes "k8s.io/client-go/kubernetes/typed/apps/v1"
+	kubeCoreTypes "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+const LatestChainlinkVersion string = "0.10.8"
+
+type Environment interface {
+	ChainlinkNodes() []client.Chainlink
+	ChainlinkNodeETHAddresses() ([]common.Address, error)
+	Adapter() ExternalAdapter
+	BlockchainClient() client.BlockchainClient
+	Wallets() client.BlockchainWallets
+	ContractDeployer() contracts.ContractDeployer
+	LinkContract() contracts.LinkToken
+
+	FundAllNodes(fromWallet client.BlockchainWallet, nativeAmount, linkAmount *big.Int) error
+	TearDown() error
+}
+
+type environment struct {
+	name                  string
+	kubeClient            *kubernetes.Clientset
+	deploymentsClient     kubeAppTypes.DeploymentInterface
+	servicesClient        kubeCoreTypes.ServiceInterface
+	network               client.BlockchainNetwork
+	chainlinkNodes        []client.Chainlink
+	forwardedPortChannels []chan os.Signal
+	adapter               ExternalAdapter
+	blockchainClient      client.BlockchainClient
+	wallets               client.BlockchainWallets
+	contractDeployer      contracts.ContractDeployer
+	linkContract          contracts.LinkToken
+}
+
+// NewBasicEnvironment launches a new environment of latest version chainlink nodes, connected to the specified network
+func NewBasicEnvironment(environmentName string, nodeCount int, network client.BlockchainNetwork) (Environment, error) {
```

Comment:
Planning to refactor this to allow cleaner, other versions, but I wanted opinions on the general approach first.